### PR TITLE
tlvf: fix int length lists parsing

### PR DESCRIPTION
**Pull request by tomereli**
- Fix bug when int length list values are initialized on parse
- Add a test to verify int len lists
- Add length function for unknown length lists
- Add unknown length list to the inner class in tlvVarList.yaml
- and a test in tlvf test to verfiy length function works as expected

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

### CODE REVIEW

File: framework/tlvf/test/main.cpp [hunk 1]
```diff
@@ -33,11 +36,66 @@ MAPF_INITIALIZE_LOGGER
 
 using namespace mapf;
 
+void dump_buffer(uint8_t *buffer, size_t len)
```
Comment (arnout) [inline framework/tlvf/test/main.cpp]: Could you make a separate commit that moves out this function from the existing code?

It might be better to return a std::string and do the LOG part in the caller. That gives more flexibility for log level, and sets the correct line and file. Eventually, I think there should be an extension to easylogging++ for hexdumps. Something like `HEXLOG(INFO, buf, buf_len);`.

File: framework/tlvf/test/main.cpp [hunk 2]
```diff
@@ -33,11 +36,66 @@ MAPF_INITIALIZE_LOGGER
...
+
+int test_int_len_list()
+{
+    int errors = 0;
+    uint8_t tx_buffer[4096];
+    const uint8_t gTlvMacAddress[6] = {0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff};
+
+    MAPF_INFO(__FUNCTION__ << " start");
+    memset(tx_buffer, 0, sizeof(tx_buffer));
+    CmduMessageTx msg = CmduMessageTx(tx_buffer, sizeof(tx_buffer));
```
Comment (arnout) [inline framework/tlvf/test/main.cpp]: For this test we don't really need a CMDU, it's sufficient to create the tlvMacAddress with the buffer as argument and call class_swap on it.

File: tools/prplMesh.code-workspace [hunk 3]
```diff
@@ -77,7 +77,10 @@
 			"utility": "cpp",
 			"atomic": "cpp",
 			"codecvt": "cpp",
-			"memory": "cpp"
+			"memory": "cpp",
+			"future": "cpp",
+			"optional": "cpp",
+			"string_view": "cpp"
```
Comment (arnout) [inline tools/prplMesh.code-workspace:83]: You probably want to commit this too, but please do it as a separate commit :-)

File: framework/tlvf/tlvf.py [hunk 4]
```diff
@@ -851,10 +851,11 @@ def addClassParam(self, obj_meta, param_name, param_type, param_type_info, param
             if (param_val_const or param_val):
                 if is_int_len or is_const_len:
```
Comment (arnout) [inline framework/tlvf/tlvf.py]: Reminder: in the other PR, I made the (nitpick) comment that these two conditions could be swapped and the is_int/const_len can be merged with the previous one, which allows to have the m_parse__ condition only once in the generated C code.

File: framework/tlvf/yaml/tlvf/test/tlvVarList.yaml [hunk 5]
```diff
@@ -38,3 +38,7 @@ cInner:
     _type: uint8_t
     _length: [ list_length ]
   var1: uint32_t
+  unknown_length_list:
```
Comment (arnout) [inline framework/tlvf/yaml/tlvf/test/tlvVarList.yaml]: As I wrote in the other PR: this can not work.

**Review by arnout — CHANGES_REQUESTED** [review]

File: framework/tlvf/tlvf.py (hunk 4 above)
Comment (tomereli) [reply framework/tlvf/tlvf.py]: Done

File: framework/tlvf/test/main.cpp [hunk 6]
```diff
@@ -24,6 +24,8 @@
 #include <mapf/common/logger.h>
 #include <tlvf/wfa_map/tlvApCapability.h>
 
+#include <algorithm>
+#include <iterator>
```
Comment (arnout) [inline framework/tlvf/test/main.cpp:29]: Neither of these headers seems to be used? I may be mistaken though. (nitpick)

File: framework/tlvf/test/main.cpp [hunk 7]
```diff
@@ -45,6 +46,39 @@ std::string dump_buffer(uint8_t *buffer, size_t len)
...
+        tlv.class_swap(); //finalize
+        LOG(DEBUG) << "TX: " << std::endl << dump_buffer(tx_buffer, tlv.getLen());
+    }
+
+    uint8_t rx_buffer[sizeof(tx_buffer)];
+    memcpy(rx_buffer, tx_buffer, sizeof(rx_buffer));
+    {
+        auto tlv = tlvMacAddress(tx_buffer, sizeof(tx_buffer), true, true);
+        auto mac = &std::get<1>(tlv.mac(0));
+        if (!std::equal(mac, mac + 6, gTlvMacAddress)) {
```
Comment (arnout) [inline framework/tlvf/test/main.cpp:71]: Ah, this is where you need `<algorithm>`.

**Review by arnout — APPROVED** [review]
Nitpicks only.

Make sure to double-check the correctness of the commit messages while rebasing.

File: framework/tlvf/test/main.cpp (hunk 6 above)
Comment (tomereli) [reply framework/tlvf/test/main.cpp:29]: Needed for std::equal
